src/docs/plugins.md

```diff
@@ -84,3 +68,51 @@ module.exports = function(eleventyConfig) {
 {% callout "warn" %}
 Plugin namespacing is an application feature and should not be used if you are creating your own plugin (in your plugin configuration code). Follow along at <a href="https://github.com/11ty/eleventy/issues/256">Issue #256</a>.
 {% endcallout %}
+
+
+## Creating a Plugin
+
+A plugin primarily provides a “configuration function.” This function is called when Eleventy is first initialized, and takes the same `eleventyConfig` object as the user’s `.eleventy.js` file gets, in addition to any config passed by the user:
+
+{% codetitle "plugin.js" %}
+
+```js
+module.exports = function(eleventyConfig, pluginOptions) {
+  // Your plugin code goes here
+};
+```
+
+If you want to run some user code before your plugin’s configuration function is run, you can instead export an object. Prefer using the above syntax unless you need this behavior. For an example of how this is used, see the [syntax highlighting plugin](https://github.com/11ty/eleventy-plugin-syntaxhighlight/blob/23761d7fd54de0312040520175959327b1a0ab9b/.eleventy.js#L10)
```

Comment:
```suggestion
If you want to run some user code before your plugin’s configuration function is run, you can instead export an object. Prefer using the above syntax unless you need this behavior. For an example of how this is used, see the [syntax highlighting plugin](https://github.com/11ty/eleventy-plugin-syntaxhighlight/blob/HEAD/.eleventy.js#L10)
```

Not sure if this PR will ever get merged, but if it did. I think it would be better to use `HEAD` instead of a specific commit SHA. `HEAD` will automatically point to the latest commit on the head branch so the URL will always stay up-to-date.

Reply:
I wanted to link to a specific commit since the line number will always stay correct; if the plugin makes unrelated changes later, that doesn’t affect the correctness of this link. (And if the method for defining plugins changes in the future, this page will need to be updated anyway)

Reply:
Smart, that didn't cross my mind actually. 👍